src/main/java/org/broadinstitute/hellbender/engine/spark/AddContextDataToReadSpark.java

```diff
@@ -25,24 +42,87 @@
  */
 public class AddContextDataToReadSpark {
     public static JavaPairRDD<GATKRead, ReadContextData> add(
-            final JavaRDD<GATKRead> reads, final ReferenceMultiSource referenceDataflowSource,
-            final JavaRDD<GATKVariant> variants, final JoinStrategy joinStrategy) {
+            final JavaSparkContext ctx,
+            final JavaRDD<GATKRead> reads, final ReferenceMultiSource referenceSource,
+            final JavaRDD<GATKVariant> variants, final JoinStrategy joinStrategy,
+            final SAMSequenceDictionary sequenceDictionary,
+            final int shardSize, final int shardPadding) {
```

Comment:
Document the new `shardSize` and `shardPadding` args, make it clear that they apply only when `JoinStrategy.OVERLAPS_PARTITIONER` is specified, and suggest values to be passed in for these args when other JoinStrategies are being used (0?).